### ISSUE TITLE
Text Field with type: "number" treated as Empty when Incrementing/Decrementing

### ISSUE THREAD

**Issue by DavidResende0** [data-driven-forms/react-forms#1037]
**Description**: When using a Text Field with the `type` property set to "number", the form will treat the field as if it's empty when entering a number by incrementing or decrementing. The form still recognizes manually entered numbers or numbers entered as an initialValue.

```
{
    component: "text-field",
    id: "number",
    isRequired: true,
    label: "Number Test",
    name: "number",
    step: 100,
    type: "number"
}
```

Manually entered number:
<img src="https://user-images.githubusercontent.com/64800041/115070346-d50a5400-9ec2-11eb-9974-d46dda2f6d32.png" width="300"> 

Incremented number:
<img src="https://user-images.githubusercontent.com/64800041/115070556-16026880-9ec3-11eb-9149-3f97f6f1c0ac.png" width="300"> 

**Schema**: https://stackblitz.com/edit/number-test?file=index.js

**Comment by rvsia** [comment]
@DavidResende0 

This is an issue on Carbon side. Check it here: https://codesandbox.io/s/awesome-rhodes-4b161

When you click on any of the arrow button, `onChange` event is called with empty `target.value`.

There have already been multiple issues opened: https://github.com/carbon-design-system/carbon/issues?q=is%3Aissue+is%3Aopen+numberInput

So, I am closing this for now. Please, feel free to reopen, if you think we can do something with it here in DDF repo.

**Comment by kavyanekkalapu** [comment]
@rvsia According to carbon docs, value is not present in e.target.value, it is in `e.imaginaryTarget.value` , check https://react.carbondesignsystem.com/?path=/docs/components-numberinput--default.

Example: https://codesandbox.io/s/misty-sea-yygnm?file=/src/index.js:463-486

**Comment by Hyperkid123** [comment]
@kavyanekkalapu seems like an easy enough change. Can you open a PR?

**Comment by kavyanekkalapu** [comment]
@Hyperkid123 Yes i can do that, but i don't see any code related to value in https://github.com/data-driven-forms/react-forms/blob/master/packages/carbon-component-mapper/src/text-field/text-field.js, so what do i need to do here, add a on change func like 
```if number input
value-->e.imaginaryTarget.value

**Comment by rvsia** [comment]
Something like this: `onChange={e => input.onChange( ... )}`

**Comment by kavyanekkalapu** [comment]
@rvsia @Hyperkid123 created https://github.com/data-driven-forms/react-forms/pull/1039, please review.

**Comment by rvsia** [comment]
Fixed in [3.5.1](https://www.npmjs.com/package/@data-driven-forms/carbon-component-mapper/v/3.5.1)

Backport to v2 released: [2.24.3](https://www.npmjs.com/package/@data-driven-forms/carbon-component-mapper/v/2.24.3)

Please, let us know if there any additional issues.
